services/galley/src/Galley/Cassandra/TeamFeatures.hs

```diff
@@ -140,13 +140,61 @@ setSelfDeletingMessagesStatus tid status = do
           <> ", self_deleting_messages_ttl) "
           <> "values (?, ?, ?)"
 
+setPaymentStatus ::
+  forall (a :: TeamFeatureName) m.
+  ( MonadClient m,
+    HasPaymentStatusCol a
+  ) =>
+  Proxy a ->
+  TeamId ->
+  PaymentStatus ->
+  m PaymentStatus
+setPaymentStatus _ tid (PaymentStatus paymentStatus) =
+  case paymentStatusCol @a of
+    Nothing -> pure $ PaymentStatus PaymentUnlocked -- todo: what should we do here? probably return an error, UpdatePaymentStatusNotAllowed or sth. similar
+    Just col -> do
+      retry x5 $ write insert (params LocalQuorum (tid, paymentStatus))
+      pure (PaymentStatus paymentStatus)
+      where
+        insert :: PrepQuery W (TeamId, PaymentStatusValue) ()
+        insert =
+          fromString $
+            "insert into team_features (team_id, "
+              <> col
+              <> ") values (?, ?)"
+
+getPaymentStatus ::
+  forall (a :: TeamFeatureName) m.
+  ( MonadClient m,
+    HasPaymentStatusCol a
+  ) =>
+  Proxy a ->
+  TeamId ->
+  m (Maybe PaymentStatus)
+getPaymentStatus _ tid =
+  case paymentStatusCol @a of
+    Nothing -> pure Nothing
```

Comment:
I would have expected `unlocked` here instead of nothing, but you can also reasonably decide to handle this further up the call tree.

Reply:
Yes, it was my intention to push this to the caller.